test/libsolidity/SolidityABIJSON.cpp

```diff
@@ -292,6 +308,8 @@ BOOST_AUTO_TEST_CASE(events)
 	{
 		"name": "f",
 		"constant": false,
+		"view": false,
```

Comment:
Memory-avoiding: `contained`, `isolated`, `confined`

I'm starting to think that we may be better with having a syntax similar to modifiers for these: `memory(changes)`, `memory(intact)`, or may be alone `memory` for those which change the memory after finishing.

Just like C inline assembly has its clobber list.